src/app/seller-post-ad-form/seller-post-ad-form.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, OnInit, ɵConsole} from '@angular/core';
+import { ImageProperty } from './imageProperty';
+import { DatePipe } from '@angular/common';
+import {NgForm} from '@angular/forms';
+
+@Component({
+  selector: 'seller-post-ad-form',
+  templateUrl: './seller-post-ad-form.component.html',
+  styleUrls: ['./seller-post-ad-form.component.css']
+})
+
+export class SellerPostAdFormComponent implements OnInit {
+  minNoOfImage=1;
+  maxNoOfImage=5;
+  isAddressSelected:boolean=false;
+  heroImageUrl = ""; // store url of hero image
+  imageUrlArray = []; // store url of images
+  imageArray:ImageProperty[] =[];
```

Comment:
Use predefined advertisement class instead of run time class decided by form model and all these properties will be part of class.